src/graph/parser.ts

```diff
@@ -294,6 +319,36 @@ export class Graph {
                         break;
                     }
 
+                    // Size field
+                    case "width":
+                    case "height": {
+                        requiresValue()
+
+                        //search for unit
+                        let index = (value as string).search(/[A-Za-z%]/)
+                        let size = { value: 0, unit: AbsoluteCSSUnit.px } as Size
+                        if (index != -1) {
+                            size.value = parseFloat((value as string).substring(0, index))
+                            let unit = (value as string).substring(index)
+                            let parseUnit = parseStringToEnum(AbsoluteCSSUnit, unit)
+                            if (parseUnit) {
+                                size.unit = parseUnit as CSSUnit
+                            } else {
+                                parseUnit = parseStringToEnum(RelativeCSSUnit, unit)
+                                if (parseUnit) {
+                                    size.unit = parseUnit as CSSUnit
+                                    graphSettings.skipCache = true
+                                } else {
+                                    throw new SyntaxError(`Unit '${unit}' must be a valid CSS Unit`);
+                                }
+                            }
+                        } else {
+                            size.value = parseFloat((value as string))
+                        }
+                        graphSettings[key] = size
```

Comment:
Could you put the bit of code that checked whether the value was a valid integer back in somewhere? It can probably just be checked after the if statement with the `size.value` variable i.e.
```typescript
if (Number.isNaN(size.value)) {
    throw new SyntaxError(`Field '${key}' must have an integer (or decimal) value`);
}
```

Reply:
ah, okay done